### PR TITLE
Cope with pagination in private zone discovery

### PR DESCRIPTION
**Pull request by Miciah**
https://github.com/openshift/cluster-ingress-operator/pull/93 changed the DNS private zone discovery to use the `GetResources` AWS API call.  However, even though we use filters, the resources are still paginated as though no filter were applied.  If the desired resource is not on the first page, then `GetResources` will not return it.  We need to use `GetResourcesPages`, and we may need to look through one or more empty pages of resources till we find the desired resource.

* `pkg/dns/aws/dns.go`: Use `GetResourcesPages` to get the private zone.

---

@ironcladlou @wking

### CODE REVIEW

**Comment by wking** [comment]
I haven't had time to go through the diff in detail, but `GetResourcesPages` is absolutely the way to go. 👍

**Comment by Miciah** [comment]
Thanks! I actually @'d you because I saw you had mentioned on an installer PR that you were following the example in this code, but now that I actually look at that PR, I see you are already using `GetResourcePages`. Sorry for the noise! Nothing to see here!

**Comment by wking** [comment]
I was riding on the "now we need the get-tagged-resources permission" coat tails ;).

**Comment by Miciah** [comment]
level=error msg="\t* module.vpc.aws_route_table_association.worker_routing[5]: 1 error occurred:"
    level=error msg="\t* aws_route_table_association.worker_routing.5: timeout while waiting for state to become 'success' (timeout: 5m0s)"

and

    could not copy stable imagestreamtag: Timeout: request did not complete within allowed duration

/retest

**Comment by Miciah** [comment]
level=error msg="\t* module.vpc.aws_route.to_nat_gw[2]: 1 error occurred:"
    level=error msg="\t* aws_route.to_nat_gw.2: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
    level=error
    level=error
    level=error msg="\t* module.vpc.aws_route.to_nat_gw[4]: 1 error occurred:"
    level=error msg="\t* aws_route.to_nat_gw.4: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
    level=error
    level=error
    level=error msg="\t* module.vpc.aws_route.to_nat_gw[3]: 1 error occurred:"
    level=error msg="\t* aws_route.to_nat_gw.3: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
    level=error
    level=error
    level=error msg="\t* module.dns.aws_route53_record.api_external: 1 error occurred:"
    level=error msg="\t* aws_route53_record.api_external: [ERR]: Error building changeset: timeout while waiting for state to become 'accepted' (timeout: 5m0s)"

and

    could not copy stable imagestreamtag: Timeout: request did not complete within allowed duration

/retest

**Comment by Miciah** [comment]
level=error msg="\t* module.vpc.aws_route.to_nat_gw[0]: 1 error occurred:"
    level=error msg="\t* aws_route.to_nat_gw.0: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"
    level=error
    level=error
    level=error msg="\t* module.vpc.aws_route.to_nat_gw[1]: 1 error occurred:"
    level=error msg="\t* aws_route.to_nat_gw.1: Error creating route: timeout while waiting for state to become 'success' (timeout: 2m0s)"

/retest

**Comment by ironcladlou** [comment]
Thank you very much!

/lgtm

File: pkg/dns/aws/dns.go [hunk 1]
```diff
@@ -135,30 +135,52 @@ func (m *Manager) discoverZones() error {
...
 			TagFilters: []*resourcegroupstaggingapi.TagFilter{
 				{
 					Key:    aws.String("openshiftClusterID"),
 					Values: []*string{aws.String(m.config.ClusterID)},
 				},
 			},
-		})
+		}, f)
+		if innerError != nil {
+			err = innerError
```
[the remark this answers is not in the document]
Comment (Miciah) [reply pkg/dns/aws/dns.go]: I figured `innerError` was more important, but yeah, aggregating the error values would be best.  I'll make a follow-up.

**Pull request merged**